astroid/interpreter/_import/spec.py

```diff
@@ -359,6 +367,28 @@ def _find_spec_with_path(
             continue
         return finder_instance, spec
 
+    # Support for custom finders
+    for meta_finder in sys.meta_path:
+        spec = meta_finder.find_spec(modname, submodule_path)
+        if spec:
+            try:
+                module_type = _MetaPathFinderModuleTypes[
+                    meta_finder.__name__  # type: ignore[attr-defined], caught by AttributeError
+                ]
+            except (KeyError, AttributeError):
+                # If we don't recognise the finder, we assume it's a regular module
+                module_type = ModuleType.PY_SOURCE
+            return (  # type: ignore[return-value]
+                meta_finder,
+                ModuleSpec(
+                    spec.name,
+                    module_type,
+                    spec.origin,
```

Comment:
Should we set location to `""` or `None` rather than `spec.origin`?

Reply:
Why? Isn't that "sort of" right?

Reply:
I think it's supposed to be `None` for namespace packages and a string for others:
https://github.com/jacobtylerwalls/astroid/commit/3ff65be6c9e3a4480cd9338a68f418cb578e347b

So we could switch on the `module_type` and change to `None` -> `""` if PY_SOURCE. But I'm interested in your opinion.

Reply:
It feels like this mimics `has_location`, see https://docs.python.org/3/library/importlib.html#importlib.machinery.ModuleSpec.has_location.

Looking for `.location` it seems like only (frozen) builtins don't have their `location` set? And we use it in `ast_from_module_name` so we need it to be set to whatever `origin` is?

Reply:
This is moot now that we're not setting `PY_SOURCE`. I was just wondering about the 0.05% of cases where we hit `KeyError` and assumed `PY_SOURCE` and then still set location to `None` instead of `""`, but it looks like after latest commits we aren't doing setting `PY_SOURCE`.